crates/pathfinder/src/core.rs

```diff
@@ -369,3 +369,71 @@ impl std::fmt::Display for Chain {
         }
     }
 }
+
+/// Adds a thin Debug and Disply implementations.
+///
+/// Debug impl produces `WrapperType(hash)` instead of automatic derives generated
+/// `WrapperType(StarkHash(hash))`.
+///
+/// Display impl produces just `hash`.
+macro_rules! thin_starkhash_debug_display {
+    ($target:ty) => {
+        thin_starkhash_debug!($target);
+
+        impl std::fmt::Display for $target {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                std::fmt::Display::fmt(&self.0, f)
+            }
+        }
+    };
+
+    ($head:ty, $($tail:ty),+ $(,)?) => {
+        thin_starkhash_debug_display!($head);
+        thin_starkhash_debug_display!($($tail),+);
+    };
+}
+
+macro_rules! thin_starkhash_debug {
+    ($target:ty) => {
+        impl std::fmt::Debug for $target {
+            fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                write!(fmt, "{}({})", stringify!($target), self.0)
+            }
+        }
+    };
+
+    ($head:ty, $($tail:ty),+ $(,)?) => {
+        thin_starkhash_debug!($head);
+        thin_starkhash_debug!($($tail),+);
+    };
+}
+
+// these types are used in sequencer tests, which require special fixed width representation
+thin_starkhash_debug!(ContractAddress, StarknetTransactionHash, ClassHash,);
```

Comment:
I've run into this a few times when refactoring. I wonder if we don't just want to fix that mock server to always strip leading zeros. Then this wouldn't be an issue.